File: frontend/src/api/geocode.ts
import { apiClient } from "./client";
import type { GeocodeResult } from "@/types/geocode";

export async function geocodeAddress(address: string): Promise<GeocodeResult> {
  const { data } = await apiClient.post<GeocodeResult>("/geocode", { address });
  return data;
}

File: backend/app/api/routes/geocode.py
from fastapi import APIRouter

from app.schemas.geocode import GeocodeRequest, GeocodeResult
from app.services.geocode_client import geocode_address

router = APIRouter(prefix="/geocode", tags=["geocode"])


@router.post("", response_model=GeocodeResult)
async def geocode(request: GeocodeRequest):
    return await geocode_address(request.address)
